src/wallet/wallet2.cpp

```diff
@@ -12901,6 +12902,74 @@ void wallet2::throw_on_rpc_response_error(const boost::optional<std::string> &st
   THROW_WALLET_EXCEPTION_IF(*status != CORE_RPC_STATUS_OK, tools::error::wallet_generic_rpc_error, method, m_trusted_daemon ? *status : "daemon error");
 }
 //----------------------------------------------------------------------------------------------------
+
+bool wallet2::save_to_file(const std::string& path_to_file, const std::string& raw) const
+{
+  if (m_export_format == ExportFormat::Hex)
+  {
+    std::string h = epee::string_tools::buff_to_hex_nodelimer(raw);
+
+    // Split the long string into multiple lines, similarly to how openssl
+    // prints certificates.
+    for (size_t i = 63; i < h.length(); i += 64)
+    {
+      h.insert(i, 1, '\n');
+    }
+
+    h.insert(h.end(), '\n');
+    
+    return epee::file_io_utils::save_string_to_file(path_to_file, h);
+  }
+  else
+  {
+    return epee::file_io_utils::save_string_to_file(path_to_file, raw);
+  }
+}
+//----------------------------------------------------------------------------------------------------
+
+bool wallet2::load_from_file(const std::string& path_to_file, std::string& target_str,
+                             size_t max_size) const
+{
+  std::string data;
+  bool r = epee::file_io_utils::load_file_to_string(path_to_file, data, max_size);
+  if (!r)
+  {
+    return false;
+  }
+
+  // Ignore any whitespace.
+  // We try to convert the version without whitespace from hex to binary.
+  // If that fails, we stick to the original data (hence the copy). 
+  std::string without_whitespace(data);
+  without_whitespace.erase(
+    std::remove_if(
+      without_whitespace.begin(), 
+      without_whitespace.end(), 
+      [](char c){ return (c =='\r' || c =='\t' || c == ' ' || c == '\n');}),
+    without_whitespace.end());
+
+  r = epee::string_tools::validate_hex(without_whitespace.size(), without_whitespace);
+  if (!r)
+  {
+    target_str = data;
+    return true;
+  }
+
+  std::string res;
+  r = epee::string_tools::parse_hexstr_to_binbuff(without_whitespace, res);
+  if (!r)
+  {
+    target_str = data;
+    return true;
+  } 
+  else
+  {
+    target_str = res;
+    return true;
```

Comment:
Could use some std::move here, the strings are likely large

Reply:
Good idea. Adding.